packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -0,0 +1,635 @@
+}
+
+/// Base class for slider thumb and value indicator shapes.
+///
+/// Create a subclass of this if you would like a custom slider thumb or
+/// value indicator shape.
+///
+/// See also:
+///
+///  * [DefaultSliderThumbShape] for a simple example of a thumb shape.
+///  * [DefaultSliderValueIndicatorShape], for a complex example of a value
+///    indicator shape.
+abstract class SliderComponentShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const SliderComponentShape();
+
+  /// Returns the preferred size of the shape, based on the given conditions.
+  Size getPreferredSize(bool isEnabled, bool isDiscrete);
+
+  /// Paints the shape, taking into account the state passed to it.
+  ///
+  /// [activationAnimation] is an animation triggered when the user beings
+  /// to interact with the slider. It reverses when the user stops interacting
+  /// with the slider.
+  /// [enableAnimation] is an animation triggered when the [Slider] is enabled,
+  /// and it reverses when the slider is disabled.
+  /// If [labelPainter] is non-null, then [labelPainter.paint] should be
+  /// called with the location that the label should appear. If the labelPainter
+  /// passed is null, then no label was supplied to the [Slider].
+  /// [value] is the current parametric value (from 0.0 to 1.0) of the slider.
+  void paint(
+    PaintingContext context,
+    bool isDiscrete,
+    Offset thumbCenter,
+    Animation<double> activationAnimation,
+    Animation<double> enableAnimation,
+    TextPainter labelPainter,
+    SliderThemeData sliderTheme,
+    TextDirection textDirection,
+    double textScaleFactor,
+    double value,
+  );
+}
+
+class DefaultSliderThumbShape extends SliderComponentShape {
+  const DefaultSliderThumbShape();
+  static const double _thumbRadius = 6.0;
+  static const double _disabledThumbRadius = 4.0;
+
+  @override
+  Size getPreferredSize(bool isEnabled, bool isDiscrete) {
+    return new Size.fromRadius(isEnabled ? _thumbRadius : _disabledThumbRadius);
+  }
+
+  @override
+  void paint(
+    PaintingContext context,
+    bool isDiscrete,
+    Offset thumbCenter,
+    Animation<double> activationAnimation,
+    Animation<double> enableAnimation,
+    TextPainter labelPainter,
+    SliderThemeData sliderTheme,
+    TextDirection textDirection,
+    double textScaleFactor,
+    double value,
+  ) {
+    final Canvas canvas = context.canvas;
+    final Tween<double> radiusTween =
+        new Tween<double>(begin: _disabledThumbRadius, end: _thumbRadius);
+    final ColorTween colorTween =
+        new ColorTween(begin: sliderTheme.disabledThumbColor, end: sliderTheme.thumbColor);
+    canvas.drawCircle(
+      thumbCenter,
+      radiusTween.evaluate(enableAnimation),
+      new Paint()..color = colorTween.evaluate(enableAnimation),
+    );
+  }
+}
+
+class DefaultSliderValueIndicatorShape extends SliderComponentShape {
```

Comment:
See earlier comment about dartdoc, not calling this DefaultFoo

Reply:
Changed to PaddleSliderValueIndicatorShape.  Kind of awkward, I know, but descriptive.